discovery-provider/integration_tests/tasks/test_index_aggregate_track.py

```diff
@@ -0,0 +1,635 @@
+
+    with db.scoped_session() as session:
+        _update_aggregate_track(session)
+
+        results: List[AggregateTrack] = (
+            session.query(AggregateTrack).order_by(AggregateTrack.track_id).all()
+        )
+
+        assert (
+            len(results) == 4
+        ), "Test that tracks updated on blocks after '1' will be targeted"
+
+        prev_id_checkpoint = get_last_indexed_checkpoint(session, AGGREGATE_TRACK)
+        assert prev_id_checkpoint == 10
+
+
+def test_index_aggregate_track_empty_completely(app):
+    """Test a completely empty database won't break"""
+
+    with app.app_context():
+        db = get_db()
+
+    entities = {}
+
+    populate_mock_db(db, entities, block_offset=3)
+
+    with db.scoped_session() as session:
+        _update_aggregate_track(session)
+
+        results: List[AggregateTrack] = (
+            session.query(AggregateTrack).order_by(AggregateTrack.track_id).all()
+        )
+
+        assert (
+            len(results) == 0
+        ), "Test that empty entities won't generate AggregateTracks"
+
+        prev_id_checkpoint = get_last_indexed_checkpoint(session, AGGREGATE_TRACK)
+        assert prev_id_checkpoint == 0
+
+
+def test_index_aggregate_track_update(app):
+    """Test that the aggregate_track data is continously added to"""
+
+    with app.app_context():
+        db = get_db()
+
+    entities = deepcopy(basic_entities)
+    entities.update(
+        {
+            "aggregate_track": [
+                {
+                    "track_id": 1,
+                    "repost_count": 9,
+                    "save_count": 9,
+                },
+                {
+                    "track_id": 2,
+                    "repost_count": 9,
+                    "save_count": 9,
+                },
+                {
+                    "track_id": 4,
+                    "repost_count": 9,
+                    "save_count": 9,
+                },
+                {
+                    "track_id": 5,
+                    "repost_count": 9,
+                    "save_count": 9,
+                },
+            ],
+        }
+    )
+
+    populate_mock_db(db, entities, block_offset=3)
+
+    with db.scoped_session() as session:
+        results: List[AggregateTrack] = (
+            session.query(AggregateTrack).order_by(AggregateTrack.track_id).all()
+        )
+
+        assert len(results) == 4, "Test that the entities were created"
+        for i, n in enumerate((0, 1, 3, 4)):
+            assert results[i].track_id == n + 1, "Test that the entities were created"
+            assert results[i].repost_count == 9, "Test that the entities were created"
+            assert results[i].save_count == 9, "Test that the entities were created"
+
+        _update_aggregate_track(session)
+
+    with db.scoped_session() as session:
+        basic_tests(session)
+
+
+def test_index_aggregate_track_update_with_extra_user(app):
+    """Test that the aggregate_track only modifies non-deleted tracks"""
```

Comment:
i believe this test case covers the following but just want to double check

- track gets deleted
- aggregate_tracks deletes the entry
- repost for the track happens (this should never be allowed client side)
- aggregate_tracks doesn't update this entry


if this is an issue, we can probably fix by adding a condition here https://github.com/AudiusProject/audius-protocol/pull/2352/files#diff-21923f3ab170287768d85b6cccbd5cb9ed1dfc2fa2e7ef2225fdc8ac2228fc32R121 like AND "track_id" NOT IN "deleted_tracks"

Reply:
Added this test. Thanks!